src/common/intrusive_red_black_tree.h

```diff
@@ -252,16 +262,18 @@ namespace impl {
 } // namespace impl
 
 template <typename T, typename Default>
-using LightCompareType = std::remove_pointer_t<decltype(impl::GetLightCompareType<T, Default>())>;
+using RedBlackKeyType =
+    typename std::remove_pointer_t<decltype(impl::GetRedBlackKeyType<T, Default>())>;
```

Comment:
```suggestion
    std::remove_pointer_t<decltype(impl::GetRedBlackKeyType<T, Default>())>;
```

SInce the typename isn't necessary with the _t variant